### DISCUSSION TITLE
Ugeplan "Not Available"

### DISCUSSION THREAD

**Discussion by c957031** [scaarup/aula#131] [General]
Først og fremmest tak til Søren for din Aula integration, jeg har meget glæde af den.

Jeg benytter navnlig skemaerne så man kan se dem uden at behøve at logge ind i Aula.

Til gengæld har jeg problemer med at få "ugeplan" til at fungere... 

Baggrundsinfo:
1.  Jeg har to børn på samme skole 
2.  Jeg kører min HA native og har installeret Aula plugin (version 0.1.35) via HACS 
3.  Følgende virker fint. Login, Aula Message, Skema
4.  Følgende virker ikke hos mig: Ugeplan
5.  Sensoren sensor.[skolenavn]_[barnets-navn] er oprettet. (I "Entiteter" findes sensoren, men ugeplan og ugeplan next er sat til "Not Available" ) se billede.
![Snip01](https://github.com/scaarup/aula/assets/77271940/11766e2b-04e7-4c8c-92f8-3fb18fdf8e2e)
6.  Jeg kan se i loggen at der tilsyneladende ikke er data i ugeplanen, omend der er data når jeg logger ind på web, minuddannelse.dk
[Aula_debug.txt](https://github.com/scaarup/aula/files/14064826/Aula_debug.txt)
7.  Jeg har nørklet lidt med koden uden at blive meget klogere...

Er der nogen der har gode ideer eller oplever det samme som jeg?
Ved I om man kan hente dokumentation for Aulas API nogen steder?

Andre gode ideer?
På forhånd tak for input...

Mvh David

**Comment by scaarup** [comment]
Hej David.

Vi mangler en linje i din log, der starter med "Widgets found:". Den dukker kun op ved opstart af HA.

Jeg mener nemlig, at der er mere end én måde at bruge ugeplaner fra "Min Uddannelse" på. Og det kunne jo være, at din skole benytter en vi ikke understøtter - i hvert fald kan vi se, fra din log, at svaret er tomt.
Men lad os se listen af widgets.

Tak.

Mvh. Søren.

**Reply by c957031** [reply]
Hej Søren,

Tak for dit svar. 

Jeg har genstartet og fundet denne linje:
[custom_components.aula.client] Widgets found: {'0049': 'Karakterer - synlig for forældre', '0023': 'MinUddannelse - SSO', '0013': 'Nyttige links', '0030': 'MinUddannelse - Opgaver', '0029': 'MinUddannelse – Ugenoter'}

Hvis der er noget jeg kan gøre for at bidrage, vil jeg gerne lægge en indsats i det.
Har du adgang til noget dokumentation man kan læse på (omkring Aula API) ?

mvh David

**Reply by scaarup** [reply]
Hej David.
Ja så ligger det data du leder efter, nok i en anden kontekst hos MU - under "'0030': 'MinUddannelse - Opgaver'" måske.

Der er ikke noget offentlig tilgængelig dokumentation for Aula's API, mig bekendt.

Det er MU's API, der skal graves i. Hvis du er skarp med developer tools i din browser, kan du se nogle af de kald der laves til MU. 

Her ses kald til MU's API når jeg er inde på aula.dk
![image](https://github.com/scaarup/aula/assets/8055470/b99a20d8-b264-4cd3-b6f1-cab901fe2b25)
Det er sådanne du skal have fanget, så kan jeg prøve at implementere det i integrationen.

**Reply by scaarup** [reply]
hov, det lader til, at vi måske er lidt heldige her: https://api.minuddannelse.net/metadata
Især her måske: https://api.minuddannelse.net/json/metadata?op=OpgavelisteRequest

Lad mig høre om du kan fange et request i din browser. Hvis ikke, kan jeg lave en custom version af integrationen hvor vi kan prøve at lave nogle kald til MU.

**Comment by scaarup** [comment]
@c957031 der er en ny branch hvor vi forsøger at lave et kald til MU Opgaver og logge svaret ud: https://github.com/scaarup/aula/tree/MU_Opgaver
Du kan evt. bare kopiere client.py.

Giv det et forsøg og lad os se loggen.

Tak.
